app/src/main/java/com/amaze/filemanager/asynchronous/services/ProgressiveServiceAbstract.java

```diff
@@ -0,0 +1,277 @@
+package com.amaze.filemanager.asynchronous.services;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.app.NotificationCompat;
+import android.text.format.Formatter;
+import android.util.Log;
+
+import com.amaze.filemanager.R;
+import com.amaze.filemanager.activities.MainActivity;
+import com.amaze.filemanager.filesystem.HybridFile;
+import com.amaze.filemanager.fragments.ProcessViewerFragment;
+import com.amaze.filemanager.ui.notifications.NotificationConstants;
+import com.amaze.filemanager.utils.DatapointParcelable;
+import com.amaze.filemanager.utils.ProgressHandler;
+import com.amaze.filemanager.utils.ServiceWatcherUtil;
+
+import java.util.ArrayList;
+
+/**
+ * @author Emmanuel Messulam <emmanuelbendavid@gmail.com>
+ *         on 28/11/2017, at 19:32.
+ */
+
+public abstract class ProgressiveServiceAbstract extends Service implements ServiceWatcherUtil.ServiceWatcherInteractionInterface {
+
+    // list of data packages which contains progress
+    public NotificationManager mNotifyManager;
+    public NotificationCompat.Builder mBuilder;
+    public int notificationID;
+    public volatile float progressPercent = 0f;
+    public ProgressHandler progressHandler;
+    public Context context;
+    // list of data packages, to initiate chart in process viewer fragment
+    public ArrayList<DatapointParcelable> dataPackages;
+    public ProgressListener progressListener;
+
+    private boolean isNotificationTitleSet = false;
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+
```

Comment:
Please stop adding new lines after the '{'. There are quite a few...

Reply:
Really? This had to come in a review?

Reply:
I've seen a LOT of these... Should I make an issue?